src/afl-cc.c

```diff
@@ -2912,7 +2916,7 @@ static void edit_params(aflcc_state_t *aflcc, u32 argc, char **argv,
     //     use. insert_param(aflcc, "-flegacy-pass-manager");
     // #endif
 
-    if (aflcc->lto_mode && !aflcc->have_c) {
+    if (aflcc->lto_mode) {
```

Comment:
why are you removing this have_c check?

Reply:
Just as what I talked a lot above:
1. Added initially in very early commit 654f389e73c9fd5b7e141b33ea28ab0fdda3178f.
2. Before #1912, `!have_c` always being true when `(lto_mode && !have_c)` being evaluated in execution, because it's used between initialized and the first set. This made the expression evaluated equally as `lto_mode`.
3. After #1912, `have_c` is 1 iff `-c` found in cmdline passed in, for each use in execution. This made LTO mode broken because we add pass, specify linker and add `lto_flag` in this `if (...)` branch. And if `-c` were found, `lto_flag` wouldn't be added.
4. This `have_c` check might be a forgotten useless trick which had been expected to be removed, IMO. Because there had been 5 `(lto_mode && !have_c)` long before, to [*try to remove warnings during compilation*](https://github.com/AFLplusplus/AFLplusplus/commit/654f389e73c9fd5b7e141b33ea28ab0fdda3178f). This is the last one left, and I can't reproduce the warnings due to this check in LTO mode during my test.

Reply:
I am not sure I understand what you write so let me explain:
if a `-c` is present, then 
```
      add_lto_linker(aflcc);
      add_lto_passes(aflcc);
```
should not be added.

Reply:
Perhaps I'm too clumsy in describing the mess🤣
Original code piece before #1912 for contrast:
![screenshot-20240109-191910](https://github.com/AFLplusplus/AFLplusplus/assets/50692172/9f65d8de-3918-4a9b-9d92-53b12a4819f3)
I just divided it into the two functions.

The key point that I can't figure out is: if 
> a `-c` is present

is really exactly equivalent to 
> `have_c` is true

why LTO mode always works well before?

AFAIK, LTO requires `-flto` regardless of whether `-c` is present or not.

Reply:
LTO always requires -flto..
have_c should always be set if -c is present and never set if -c is not present. 
lto plugin and lto linker information should only be set if NOT have_c